apollo-runtime/src/main/java/com/apollographql/apollo/internal/RealApolloCall.java

```diff
@@ -383,14 +386,24 @@ private ApolloInterceptorChain prepareInterceptorChain(Operation operation) {
     List<ApolloInterceptor> interceptors = new ArrayList<>();
 
     for (ApolloInterceptorFactory factory : applicationInterceptorFactories) {
-      interceptors.add(factory.newInterceptor());
+      ApolloInterceptor interceptor = factory.newInterceptor(logger, operation);
+      if (interceptor != null) {
+        interceptors.add(interceptor);
+      }
     }
     interceptors.addAll(applicationInterceptors);
 
     interceptors.add(responseFetcher.provideInterceptor(logger));
     interceptors.add(new ApolloCacheInterceptor(apolloStore, responseFieldMapper, dispatcher, logger));
-    if (operation instanceof Query && enableAutoPersistedQueries) {
-      interceptors.add(new ApolloAutoPersistedQueryInterceptor(logger, useHttpGetMethodForPersistedQueries));
+    if (autoPersistedOperationsInterceptorFactory != null) {
+      ApolloInterceptor interceptor = autoPersistedOperationsInterceptorFactory.newInterceptor(logger, operation);
+      if (interceptor != null) {
+        interceptors.add(interceptor);
+      }
+    } else {
+      if (operation instanceof Query && enableAutoPersistedQueries) {
```

Comment:
So it seems like this choose to preserve the old behavior of not allowing persisted mutations if you don't use a custom interceptor? Fine with me since we plan to use  a custom one :) but is there plan to follow-up to investigate enabling this for everyone?

Reply:
The plan would be to encourage using the custom interceptor approach for more advanced use cases. Maybe even deprecate `enableAutoPersistedQueries` and `useHttpGetMethodForPersistedQueries`. There's a lot of boilerplate involved right now in passing these variables over from the `ApolloClientBuilder` to `ApolloClient` to `RealApolloCall` (and even `QueryRefetcher`) and I didn't really fell like adding a new `enableAutoPersistedMutations` boolean. It _feels_ like most of this state can be local to the interceptor. Also, `ApolloAutoPersistedOperationInterceptor` is public so that users won't have to reimplement a full interceptor but can do something like:

```kotlin
builder.setAutoPersistedOperationsInterceptorFactory(ApolloAutoPersistedOperationInterceptor.Factory())
```